src/main/java/com/shrralis/ssdemo1/security/handler/CitizenAuthenticationFailureHandler.java

```diff
@@ -32,9 +36,22 @@ public void onAuthenticationFailure(HttpServletRequest request,
 		if (e.getClass().isAssignableFrom(UsernameNotFoundException.class)) {
 			logger.error("UsernameNotFoundException: {}", e);
 			MAPPER.writeValue(response.getWriter(), new JsonResponse(JsonError.Error.USER_NOT_EXIST.forField("login")));
+		} else if (e.getClass().isAssignableFrom(EmailNotFoundException.class)) {
```

Comment:
Look at strategy and factory method patterns combination

Reply:
I've changed that, look in coming commit, please.